### PR TITLE
[bazel] Enable C warning flags in .bazelrc

### LINKED ISSUE #12553: [bazel] Warning flags
Looking at the compiler invocations [here](https://docs.google.com/document/d/1p3Yqj4-iupAnJGPCzI_P3Pbc8X3hcCJl56D5MvbBJIo/edit#heading=h.fjx8rqc2m3up) (thanks @mcy!), bazel seems to be missing several warnings. We should have the following warnings to keep parity with meson builds:

```
  -Wall
  -Werror
  -Wextra
  -Wgnu
  -Wimplicit-fallthrough
  -Winvalid-pch
  -Wno-covered-switch-default
  -Wno-error=unused-function
  -Wno-gnu-zero-variadic-macro-arguments
  -Wno-missing-field-initializers
  -Wno-sign-compare
  -Wno-unused-parameter
  -Wstrict-prototypes
  -Wswitch-default
```

### PR DESCRIPTION
**Pull request by dmcardle**
Fixes #12553

### CODE REVIEW

**Review by alphan — APPROVED** [review]
Thanks @dmcardle! It would be great if we could minimize new `OT_UNREACHABLE()`s.
Also, it would be a good idea to rebase this PR as you suggested.

**Comment by dmcardle** [comment]
> Thanks @dmcardle! It would be great if we could minimize new `OT_UNREACHABLE()`s. Also, it would be a good idea to rebase this PR as you suggested.

Done! Managed to get rid of the new `OT_UNREACHABLE`s by zero-initializing local variables.

File: sw/device/tests/spi_host_smoketest.c [hunk 1]
```diff
@@ -155,7 +155,7 @@ status_t test_quad_read(void) {
 static bool is_4_bytes_address_mode_supported(void) {
   enum { kSupportOnly3Bytes, kSupport3and4Bytes, kSupportOnly4Bytes };
   uint8_t address_mode =
-      bitfield_field32_read(sfdp.bfpt[0], SPI_FLASH_ADDRESS_MODE);
+      (uint8_t)bitfield_field32_read(sfdp.bfpt[0], SPI_FLASH_ADDRESS_MODE);
```
Comment (engdoreis) [inline sw/device/tests/spi_host_smoketest.c:158]: I would just change the type of `address_mode`.

**Review by engdoreis — APPROVED** [review]

File: sw/device/tests/spi_host_smoketest.c (hunk 1 above)
Comment (dmcardle) [reply sw/device/tests/spi_host_smoketest.c]: Done!

**Comment by dmcardle** [comment]
@msfschaffner I'm sorry if this is causing headaches. Feel free to revert if it's not feasible to fix-forward. This was the last wide-reaching change that modified C warning flags.

What could we do to prevent this from happening again? Can we add a test to CI?

I'm also not 100% sure which changes you're asking to be notified about — changes to C flags?

**Comment by msfschaffner** [comment]
Let's leave it in for now - we'll attempt to fix this since we should minimize differences between the open and closed DV environments. The simplest way to check this in CI is likely to add this and similar global config files to the CI blockfile.

I know it's not always obvious (I ran into this issue myself) - but we should be mindful about changes that are known to have a global impact on the DV environment. The closed-source side is using the same infrastructure and adds additional C tests on top that are not exposed in the open-source.

**Comment by jdonjdon** [comment]
@dmcardle, @msfschaffner 
I am little late for this thread. because I was fixing compile errors in the closed source.
Changes added to .bazelrc cause about 12 regression failures in closed source regression.
I fixed all closed source files for now, but at least for the next a few weeks,
we need a stable rule check and focus our time to fix run_time test failure.

**Pull request merged**